src/fides/api/ops/service/connectors/saas/authenticated_client.py

```diff
@@ -76,27 +82,80 @@ def get_authenticated_request(
         # otherwise just return the prepared request
         return req
 
+    def retry_send(  # type: ignore
+        retry_count: int,
+        backoff_factor: float,
+        retry_status_codes: List[int] = [429, 502, 503, 504],
+    ) -> Callable:
+        """
+        Retry decorator for http requests, backing off exponentially or listening to server retry-after header
+
+        Exponential backoff factor uses the following formula:
+        backoff_factor * (2 ** (retry_attempt))
+        For an backoff_factor of 1 it will sleep for 2,4,8 seconds
+
+        General exceptions are not retried. RequestFailureResponseException exceptions are only retried
+        if the status code is in retry_status_codes.
+        """
+
+        def decorator(func: Callable) -> Callable:
+            @wraps(func)
+            def result(*args: Any, **kwargs: Any) -> Response:
+                self = args[0]
+                last_exception: Optional[Union[BaseException, Exception]] = None
+
+                for attempt in range(retry_count + 1):
+                    sleep_time = backoff_factor * (2 ** (attempt + 1))
+                    try:
+                        return func(*args, **kwargs)
+                    except RequestFailureResponseException as exc:  # pylint: disable=W0703
+                        response: Response = exc.response
+                        status_code: int = response.status_code
+                        last_exception = ClientUnsuccessfulException(
+                            status_code=status_code
+                        )
+
+                        if status_code not in retry_status_codes:
+                            break
+
+                        # override sleep time if retry after header is found
+                        retry_after_time = get_retry_after(response)
+                        sleep_time = (
+                            retry_after_time if retry_after_time else sleep_time
+                        )
+                    except Exception as exc:  # pylint: disable=W0703
+                        dev_mode_log = f" with error: {exc}" if config.dev_mode else ""
+                        last_exception = ConnectionException(
+                            f"Operational Error connecting to '{self.key}'{dev_mode_log}"
+                        )
+                        # requests library can raise ConnectionError, Timeout or TooManyRedirects
+                        # we will not retry these as they don't usually point to intermittent issues
+                        break
+
+                    if attempt < retry_count:
+                        logger.warning(
+                            "Retrying http request in %s seconds", sleep_time
+                        )
+                        sleep(sleep_time)
+
+                raise last_exception  # type: ignore
+
+            return result
+
+        return decorator
+
+    @retry_send(retry_count=3, backoff_factor=1.0)  # pylint: disable=E1124
```

Comment:
Hey @Kelsey-Ethyca, we have similar systems for retrying nodes in the traversal albeit with one major difference, that we're able to set the values with [these config. variables](https://github.com/ethyca/fides/blob/unified-fides-2/.fides/fides.toml#L47-L48). This comes with the added benefit of being able to moderate the values during testing — ie. the tests don't waste time waiting around for requests to retry that are always destined to fail.

Can we fast-track a change for this please?